Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h

```diff
@@ -41,24 +43,22 @@
 #define DISABLE_JTAG
 
 //
-// Enable SD EEPROM to prevent infinite boot loop
+// EEPROM
 //
-#ifdef ARDUINO_ARCH_STM32
-  #define FLASH_EEPROM_EMULATION
-  #define EEPROM_PAGE_SIZE     (0x800U) // 2KB
-  #define EEPROM_START_ADDRESS (0x8000000UL + (STM32_FLASH_SIZE) * 1024UL - (EEPROM_PAGE_SIZE) * 2UL)
-  #define MARLIN_EEPROM_SIZE (EEPROM_PAGE_SIZE)
-#else
-  #define SDCARD_EEPROM_EMULATION
+#if NO_EEPROM_SELECTED
+  #define IIC_EEPROM_SDA                    PB7
+  #define IIC_EEPROM_SCL                    PB6
```

Comment:
Marlin only needs these two pin definitions when using the BL24C16 chip included on Creality 4.x boards. If this board has a regular wired (I2C or SPI) EEPROM, the better choice would be `I2C_EEPROM` or `SPI_EEPROM`. If it doesn't have one of these then Flash-emulated EEPROM is preferred so that the SD card can be removed without any consequence.

Reply:
The original MKS Robin board have two external storage devices:
- The AT24C16N-10SI-2.4 16Kb (2KB) I2C EEPROM (but some variants seems to have unknown chinese clones).
- The W25Q64BVSSI 64Kb (8KB) SPI FLASH.

My goal was to make use of the I2C EEPROM as i don't know what can marlin do with a SPÏ flash except using it as an emulated EEPROM. I tried to simply use I2C_EEPROM and it compiled just fine, but marlin was frozen on the boot screen as if no EEPROM was detected. After hours of searching, i tried to use the Creality way of defining the I2C EEPROM and it worked just fine. I don't know what exactly this changes in the behavior of marlin, according to you it seems to be a way to communicate with EEPROM chips that have been weirdly wired ans maybe this is the case for the Robin v2.2 I have as it works just fine with this (and it also seems to be the case for @mattdog01 Robin v2.3).

I also added a pin definition for the W25Q64BVSSI SPI flash and it compiles. I don't know if it works as my marlin don't use the SPI flash chip. If anyone knows how to make an interesting use of this chip (such as deporting ColorUI assets to it for example...) i'd be really interested.

Reply:
I think @rhapsodyv was doing SPI flash work, although in his case it was flash on a display PCB and not on the motherboard itself.

Reply:
We use spi flash to store LVGL icons. We use about 1mb to 2mb.